src/marshmallow/fields.py

```diff
@@ -855,16 +868,14 @@ def _deserialize(self, value, attr, data, **kwargs) -> typing.Any:
             raise self.make_error("invalid_utf8") from error
 
 
-class UUID(String):
+class UUID(Field[uuid.UUID]):
```

Comment:
UUID inheriting from String was always a suboptimal choice, as was suggested in https://github.com/marshmallow-code/marshmallow/pull/1485#issuecomment-579922293

> As I can see, fields inheriting from String, are in general deserialized from strings to strings with additional input value format validation. We can see it in the Url and Email fields cases. Exception from this rule is the UUID field, which deserializes from string into UUID object representation. Therefore, it is not clear for me why UUID inherits from String and not Field, like in the case of DateTime or TimeDelta which also deserializes to object representation.